File: python/exercises/code-smells/feature-envy/src/customer.py
from dataclasses import dataclass


@dataclass
class Customer:
    id: str
    name: str
    email: str
    type: str
    loyalty_years: int
    address: str = ""
    phone_number: str = ""

    def get_id(self) -> str:
        return self.id

    def get_name(self) -> str:
        return self.name

    def get_email(self) -> str:
        return self.email

    def get_type(self) -> str:
        return self.type

    def get_loyalty_years(self) -> int:
        return self.loyalty_years

    def get_address(self) -> str:
        return self.address

    def get_phone_number(self) -> str:
        return self.phone_number

File: typescript/exercises/code-smells/feature-envy/src/Customer.ts
export class Customer {
  constructor(
    private readonly id: string,
    private readonly name: string,
    private readonly email: string,
    private readonly type: string,
    private readonly loyaltyYears: number,
    private readonly address: string = '',
    private readonly phoneNumber: string = ''
  ) {}

  getId(): string {
    return this.id;
  }

  getName(): string {
    return this.name;
  }

  getEmail(): string {
    return this.email;
  }

  getType(): string {
    return this.type;
  }

  getLoyaltyYears(): number {
    return this.loyaltyYears;
  }

  getAddress(): string {
    return this.address;
  }

  getPhoneNumber(): string {
    return this.phoneNumber;
  }
}
